src/core/include/openvino/op/search_sorted.hpp

```diff
@@ -22,7 +22,15 @@ class OPENVINO_API SearchSorted : public Op {
     /// \param values          Values to search indexs for.
     /// \param right_mode      If False, return the first suitable index that is found for given value. If True, return
     /// the last such index.
-    SearchSorted(const Output<Node>& sorted_sequence, const Output<Node>& values, bool right_mode = false);
+    /// \param output_element_type The element type of the output tensor. This is purely an implementation flag, which
+    /// is needed to enable some transformation in ConvertPrecision for CPU plugin(and potentially other plugins as
+    /// well). Setting this flag to element::i32 will result in the output tensor of i32 element type. Setting this flag
+    /// to element::i64 will generally not give any effect, since it will be converted to i32 anyway, at least for CPU
+    /// plugin.
+    SearchSorted(const Output<Node>& sorted_sequence,
+                 const Output<Node>& values,
+                 bool right_mode = false,
+                 const element::Type& output_element_type = element::i64);
```

Comment:
minor:
I would recommend to use 
"output_type" instead of "output_element_type" to be align with ShapeOf, NMS ops:

https://github.com/openvinotoolkit/openvino/blob/e07546d6414768c7fea3769eef40bcb544e9352a/src/core/src/op/shape_of.cpp#L112
https://github.com/openvinotoolkit/openvino/blob/e07546d6414768c7fea3769eef40bcb544e9352a/src/core/src/op/non_max_suppression.cpp#L329

This will make it easier to find all places with that attribute

Reply:
Fixed.

Reply:
It makes the node `set_output_type` from the base Node class, and the setter for attribute ambigous.
So it's ok to name it slightly different.
cc: @praasz

Reply:
PythonAPI generates setters automatically based on the attribute names, so `output_type` name of the attribute makes it impossible to distinguish setter from the Node class method, and we had some issues with that in the mentioned ops like NMS.
The `output_type` as a name of the attribute makes things more complicated and can lead to some unexpected behavior from the PythonAPI user perspective. But as the fix is needed asap let's make a final decision for this attr name separately.